src/components/RegistrationLandingPage.tsx

```diff
@@ -50,50 +40,15 @@ export const RegistrationLandingPage = () => {
             Email: <span className="text-gray-500 ml-10"> {email} </span>
           </span>
         </div>
-        <div className="mb-4">
-          <span className="text-primary-font-color">Hasło: </span>
-          <Controller
-            control={control}
-            name="password"
-            render={({ field }) => (
-              <input
-                type="password"
-                onChange={field.onChange}
-                onBlur={field.onBlur}
-                value={field.value}
-                name={field.name}
-                placeholder="Podaj hasło"
-                className="w-full p-2 bg-secondary-color text-primary-placeholder-font-color border border-primary-border-color mt-2"
-              />
-            )}
-          />
-          {errors.password?.message && <span className="text-red-500">{errors.password.message}</span>}
-        </div>
-        <div className="mb-4">
-          <span className="text-primary-font-color">Powtórz hasło: </span>
-          <Controller
-            control={control}
-            name="repeatPassword"
-            render={({ field }) => (
-              <input
-                type="password"
-                onChange={field.onChange}
-                onBlur={field.onBlur}
-                value={field.value}
-                name={field.name}
-                className="w-full p-2 bg-secondary-color text-primary-placeholder-font-color border border-primary-border-color mt-2"
-                placeholder="Powtórz hasło"
-              />
-            )}
-          />
-          {errors.repeatPassword?.message && <span className="text-red-500">{errors.repeatPassword.message}</span>}
-        </div>
-        <button
-          className="font-sans bg-login-btn-color w-full p-2 bg-red-500 text-primary-font-color hover:bg-red-600 transition-colors duration-300 mt-4"
-          type="submit"
-        >
-          Zarejestruj się
-        </button>
+        <DefaultInput type="password" control={control} errors={errors} name="password" placeholder="Hasło" />
+        <DefaultInput
+          type="password"
+          control={control}
+          errors={errors}
+          name="repeatPassword"
+          placeholder="Powtórz hasło"
+        />
+        <RedButton buttonType="submit" buttonMessage="Zarejestruj się" link="/dashboard" />
```

Comment:
Przez to, że jest podpięty `Link` w przycisku nie byłem w stanie w ogóle sprawdzić działania `react-hook-forms` (chodzi o błędy).